### ISSUE TITLE
Documentation should state version described

### ISSUE THREAD

**Issue by gerritholl** [pytroll/satpy#1225]
## Feature Request

**Is your feature request related to a problem? Please describe.**

The [documentation on Readthedocs](https://satpy.readthedocs.io/en/latest/) doesn't state what version of Satpy is documented.  This is potentially problematic, because it means the reader cannot easily verify whether the online documentation matches the version they have installed.  There is a small dropdown menu at the bottom left saying "latest", which can be changed to "stable".  Presumably "latest" is the current master and "stable" is the most recent release, but this is not stated anywhere as far as I can see, and the user cannot easily verify that readthedocs documentation have indeed been recently updated.  There is also a revision (presumably a git revision) at the bottom, but that is not a very user-friendly way to communicate a version.

**Describe the solution you'd like**

I would like that the documentation states what version it relates to, preferably in a footer so that it's visible on every page.  Perhaps there could also be a permanent link such that apart from `/en/latest` and `/en/stable` the user can navigate to `/en/v0.21` to be sure they're looking at the right version.

**Describe any changes to existing user workflow**

No changes.

**Additional context**

Applies to other Pytroll projects as well.

**Comment by djhoese** [comment]
Good point. Overall I agree. I think this is one of those common practice things in the scientific python ecosystem (or maybe the larger python ecosystem) that sphinx docs typically have a latest and a stabe and they are as you've described. It is possible for us to keep builds for past versions, but since things were changing so much and so quickly it wasn't really considered. I'm not sure if we have a limit for number of builds we can hold on to either.

Bottom line: This is possible, but we'll have to come up with an easy to maintain option.
